### PR TITLE
unit tests use tas pipeline

### PR DESCRIPTION
**Pull request by miyunari**

### CODE REVIEW

File: .tekton/cosign-pull-request.yaml [hunk 1]
```diff
@@ -176,8 +162,6 @@ spec:
         values:
         - "true"
       workspaces:
-      - name: output
-        workspace: workspace
       - name: basic-auth
         workspace: git-auth
     - name: prefetch-dependencies
```
Comment (JasonPowr) [inline .tekton/cosign-pull-request.yaml:167]: We need to add extra parameters here:
```
    - name: prefetch-dependencies
      params:
      - name: input
        value: $(params.prefetch-input)
      - name: hermetic
        value: ${params.hermetic}
      - name: SOURCE_ARTIFACT
        value: $(tasks.clone-repository.results.SOURCE_ARTIFACT)
      - name: ociStorage
        value: $(params.output-image).prefetch
      - name: ociArtifactExpiresAfter
        value: $(params.image-expires-after)
```

File: .tekton/cosign-pull-request.yaml [hunk 2]
```diff
@@ -189,20 +173,12 @@ spec:
...
           value: task
         resolver: bundles
-      when:
-      - input: $(params.hermetic)
-        operator: in
-        values:
-        - "true"
-      workspaces:
-      - name: source
-        workspace: workspace
```
Comment (JasonPowr) [inline .tekton/cosign-pull-request.yaml:205]: We just need to remove the workspace here:

File: .tekton/cosign-pull-request.yaml [hunk 3]
```diff
@@ -215,18 +191,30 @@ spec:
...
       - name: PREFETCH_INPUT
         value: $(params.prefetch-input)
+      - name: hermetic
+        value: ${params.hermetic}
+      - name: SOURCE_ARTIFACT
+        value: $(tasks.clone-repository.results.SOURCE_ARTIFACT)
+      - name: ociStorage
+        value: $(params.output-image).prefetch
+      - name: ociArtifactExpiresAfter
+        value: $(params.image-expires-after)
```
Comment (JasonPowr) [inline .tekton/cosign-pull-request.yaml:201]: These parameters are not needed here

File: .tekton/cosign-pull-request.yaml [hunk 4]
```diff
@@ -326,14 +312,9 @@ spec:
...
         values:
         - "false"
-      workspaces:
-      - name: workspace
-        workspace: workspace
       params:
-      - name: image-digest
-        value: $(tasks.build-container.results.IMAGE_DIGEST)
-      - name: image-url
-        value: $(tasks.build-container.results.IMAGE_URL)
```
Comment (JasonPowr) [inline .tekton/cosign-pull-request.yaml:336]: We can add these parameters back

File: .tekton/cosign-pull-request.yaml [hunk 5]
```diff
@@ -378,44 +359,21 @@ spec:
...
+    - name: run-unit-test
+      runAfter:
+      - prefetch-dependencies
+      taskRef:
+        name: go-unit-test
+      params:
+        - name: SOURCE_ARTIFACT
+          value: $(tasks.prefetch-dependencies.results.SOURCE_ARTIFACT)
+        - name: CACHI2_ARTIFACT
+          value: $(tasks.prefetch-dependencies.results.CACHI2_ARTIFACT)
```
Comment (JasonPowr) [inline .tekton/cosign-pull-request.yaml:371]: Nit pick, the params here are indented one to many

File: .tekton/cosign-push.yaml [hunk 6]
```diff
@@ -323,14 +309,9 @@ spec:
...
         values:
         - "false"
-      workspaces:
-      - name: workspace
-        workspace: workspace
       params:
-      - name: image-digest
-        value: $(tasks.build-container.results.IMAGE_DIGEST)
-      - name: image-url
-        value: $(tasks.build-container.results.IMAGE_URL)
```
Comment (JasonPowr) [inline .tekton/cosign-push.yaml:333]: These can be added back

File: .tekton/cosign-push.yaml [hunk 7]
```diff
@@ -375,44 +356,21 @@ spec:
...
+    - name: run-unit-test
+      runAfter:
+      - prefetch-dependencies
+      taskRef:
+        name: go-unit-test
+      params:
+        - name: SOURCE_ARTIFACT
+          value: $(tasks.prefetch-dependencies.results.SOURCE_ARTIFACT)
+        - name: CACHI2_ARTIFACT
+          value: $(tasks.prefetch-dependencies.results.CACHI2_ARTIFACT)
```
Comment (JasonPowr) [inline .tekton/cosign-push.yaml:368]: same as above, indention is one too many

File: .tekton/cosign-unit-test.yaml [hunk 8]
```diff
@@ -5,16 +5,46 @@ metadata:
...
+        fi
         go mod vendor
         go test ./...
+  volumes:
+    - name: workdir
+      emptyDir: {}
+    - name: cachi2
+      emptyDir: {}
 
 # This file bundles the unit tests for cosign.
```
Comment (JasonPowr) [inline .tekton/cosign-unit-test.yaml:49]: These comments can be removed

**Review by JasonPowr — CHANGES_REQUESTED** [review]
Found some issue that may be stopping the pipeline from running :) ptal

File: .tekton/cosign-pull-request.yaml (hunk 2 above)
Comment (miyunari) [reply .tekton/cosign-pull-request.yaml:205]: ups :D that was a mistake^^

**Comment by miyunari** [comment]
> Found some issue that may be stopping the pipeline from running :) ptal

Perfect, thanks a lot! I adressed everything you found :+1:

**Comment by miyunari** [comment]
@JasonPowr,  could you review the PR again? :) I think all the remaining issues seen in the ci-pipeline are not related to my work - wdyt? :)

**Comment by JasonPowr** [comment]
@miyunari This looks good to me, just one small thing, can you rebase this pr, the `sbom-json-check` task was recently removed in this commit: https://github.com/securesign/cosign/pull/245/commits/98d3f7e4e52836d993699b7b5ec17e72ded8bab9, this is due to the fact that the task is being deprecated, then I think you should be all good :) 

https://github.com/securesign/cosign/pull/247/files#diff-73cfefa52edbf682c98b967f363b95a50608bb56f412d4c96acfddca4535c442R341

**Review by JasonPowr — APPROVED** [review]
/lgtm

**Pull request merged**